hydra-node/src/Hydra/Chain/Direct/Tx.hs

```diff
@@ -569,50 +501,70 @@ observeCommit networkId tx = \case
 -- and decoding its redeemer.
 observeCollectComTx ::
   -- | A Utxo set to lookup tx inputs
-  Map (TxIn StandardCrypto) (TxOut Era) ->
+  UTxO Era ->
   ValidatedTx Era ->
   Maybe (OnChainTx CardanoTx, OnChainHeadState)
 observeCollectComTx utxo tx = do
-  (headInput, headOutput) <- findScriptOutput utxo headScript
-  redeemer <- getRedeemerSpending tx headInput
-  oldHeadDatum <- lookupDatum (wits tx) headOutput
+  (headInput, headOutput) <- Api.findScriptOutput @Api.PlutusScriptV1 (Api.fromLedgerUtxo utxo) headScript
+  redeemer <-
+    Api.findRedeemerSpending
+      (Api.getTxBody $ Api.fromLedgerTx tx)
+      headInput
```

Comment:
Nitpick: EOLs seem inconsistent between the 2 statements here (L508 and 509-512)

Reply:
Suggestion: Create a single function that returns everything we need in one go (input, output, datum, redeemer)?